pkg/clusteragent/clusterchecks/stores.go

```diff
@@ -52,12 +52,16 @@ func (s *clusterStore) getNodeStore(nodeName string) (*nodeStore, bool) {
 
 // getOrCreateNodeStore retrieves the store struct for a given node name.
 // If the node is not yet in the store, an entry will be inserted and returned.
-func (s *clusterStore) getOrCreateNodeStore(nodeName string) *nodeStore {
+func (s *clusterStore) getOrCreateNodeStore(nodeName, clientIP string) *nodeStore {
 	node, ok := s.nodes[nodeName]
 	if ok {
+		if node.clientIP != clientIP && clientIP != "" {
+			log.Debugf("Client IP changed for node %s: updating %s to %s", nodeName, node.clientIP, clientIP)
+			node.clientIP = clientIP
```

Comment:
Do you need to `node.Lock` this ?

Reply:
I don't think so, we already lock the `store` when calling this function, also we made it clean in the `clusterStore` and `nodeStore` struct definitions that only the dispatcher must hold the lock.

```
// nodeStore holds the state store for one node.
// Lock is to be held by the user (dispatcher)
type nodeStore struct {
...
```

```
// clusterStore holds the state of cluster-check management.
// Lock is to be held by the dispatcher so it can make atomic
// operations involving several calls.
type clusterStore struct {
...
```